src/Proto3/Wire/Builder.hs

```diff
@@ -385,10 +390,31 @@ word64LE w = Builder (Sum 8, BB.word64LE w)
+    else if i < 2097152 then Builder (Sum 3) (BB.word8 (fromIntegral i .|. 0x80) <> BB.word8 (fromIntegral (i `shiftR` 7) .|. 0x80) <> BB.word8 (fromIntegral (i `shiftR` 14)))
+    else if i < 268435456 then Builder (Sum 4) (BB.word8 (fromIntegral i .|. 0x80) <> BB.word8 (fromIntegral (i `shiftR` 7) .|. 0x80) <> BB.word8 (fromIntegral (i `shiftR` 14) .|. 0x80) <> BB.word8 (fromIntegral (i `shiftR` 21)))
+    else if i < 34359738368 then Builder (Sum 5) (BB.word8 (fromIntegral i .|. 0x80) <> BB.word8 (fromIntegral (i `shiftR` 7) .|. 0x80) <> BB.word8 (fromIntegral (i `shiftR` 14) .|. 0x80) <> BB.word8 (fromIntegral (i `shiftR` 21) .|. 0x80) <> BB.word8 (fromIntegral (i `shiftR` 28)))
+    else if i < 4398046511104 then Builder (Sum 6) (BB.word8 (fromIntegral i .|. 0x80) <> BB.word8 (fromIntegral (i `shiftR` 7) .|. 0x80) <> BB.word8 (fromIntegral (i `shiftR` 14) .|. 0x80) <> BB.word8 (fromIntegral (i `shiftR` 21) .|. 0x80) <> BB.word8 (fromIntegral (i `shiftR` 28) .|. 0x80) <> BB.word8 (fromIntegral (i `shiftR` 35)))
+    else if i < 562949953421312 then Builder (Sum 7) (BB.word8 (fromIntegral i .|. 0x80) <> BB.word8 (fromIntegral (i `shiftR` 7) .|. 0x80) <> BB.word8 (fromIntegral (i `shiftR` 14) .|. 0x80) <> BB.word8 (fromIntegral (i `shiftR` 21) .|. 0x80) <> BB.word8 (fromIntegral (i `shiftR` 28) .|. 0x80) <> BB.word8 (fromIntegral (i `shiftR` 35 .|. 0x80)) <> BB.word8 (fromIntegral (i `shiftR` 42)))
+    else if i < 72057594037927936 then Builder (Sum 8) (BB.word8 (fromIntegral i .|. 0x80) <> BB.word8 (fromIntegral (i `shiftR` 7) .|. 0x80) <> BB.word8 (fromIntegral (i `shiftR` 14) .|. 0x80) <> BB.word8 (fromIntegral (i `shiftR` 21) .|. 0x80) <> BB.word8 (fromIntegral (i `shiftR` 28) .|. 0x80) <> BB.word8 (fromIntegral (i `shiftR` 35 .|. 0x80)) <> BB.word8 (fromIntegral (i `shiftR` 42) .|. 0x80) <> BB.word8 (fromIntegral (i `shiftR` 49)))
+    else if i < 9223372036854775808 then Builder (Sum 9) (BB.word8 (fromIntegral i .|. 0x80) <> BB.word8 (fromIntegral (i `shiftR` 7) .|. 0x80) <> BB.word8 (fromIntegral (i `shiftR` 14) .|. 0x80) <> BB.word8 (fromIntegral (i `shiftR` 21) .|. 0x80) <> BB.word8 (fromIntegral (i `shiftR` 28) .|. 0x80) <> BB.word8 (fromIntegral (i `shiftR` 35 .|. 0x80)) <> BB.word8 (fromIntegral (i `shiftR` 42) .|. 0x80) <> BB.word8 (fromIntegral (i `shiftR` 49) .|. 0x80) <> BB.word8 (fromIntegral (i `shiftR` 56)))
+    else                                Builder (Sum 10) (BB.word8 (fromIntegral i .|. 0x80) <> BB.word8 (fromIntegral (i `shiftR` 7) .|. 0x80) <> BB.word8 (fromIntegral (i `shiftR` 14) .|. 0x80) <> BB.word8 (fromIntegral (i `shiftR` 21) .|. 0x80) <> BB.word8 (fromIntegral (i `shiftR` 28) .|. 0x80) <> BB.word8 (fromIntegral (i `shiftR` 35 .|. 0x80)) <> BB.word8 (fromIntegral (i `shiftR` 42) .|. 0x80) <> BB.word8 (fromIntegral (i `shiftR` 49) .|. 0x80) <> BB.word8 (fromIntegral (i `shiftR` 56) .|. 0x80) <> BB.word8 (fromIntegral (i `shiftR` 63)))
```

Comment:
It might be easier to read this kind of code if it put each octet on its own line.

Reply:
much nicer!